config/initializers/014-track-setting-changes.rb

```diff
@@ -87,4 +87,16 @@
   end
 
   Theme.expire_site_cache! if name == :default_theme_id
+
+  if name == :content_localization_enabled && new_value == true
+    %i[post_menu post_menu_hidden_items].each do |setting_name|
+      current_items = SiteSetting.public_send(setting_name).split("|")
+      if current_items.exclude?("addTranslation")
+        edit_index = current_items.index("edit")
+        insert_position = edit_index ? edit_index + 1 : 0
+        current_items.insert(insert_position, "addTranslation")
+        SiteSetting.public_send("#{setting_name}=", current_items.join("|"))
```

Comment:
```suggestion
        SiteSetting.set(setting_name, current_items.join("|"))
```
We have `SiteSetting.set` too.